resources/repo.rb

```diff
@@ -0,0 +1,90 @@
+property :repo_name, String, name_attribute: true
+property :repo_type, String, default: 'maven-hosted'
+property :config, Hash
+property :api_url, String
+property :api_user, String
+property :api_password, String
+
+def apiclient
+  @apiclient ||= Nexus3::Api.new(api_url, api_user, api_password)
+end
+
+def textify(args)
+  require 'json'
+  JSON.generate(args)
+end
+
+load_current_value do |new_resource|
+  config_as_json = apiclient.run_script('get_repo',
+                                        textify(name: current_resource.repo_name, type: current_resource.repo_type))
+  config = JSON.parse(config_as_json)
+end
+
+action :create do
+  nexus_api 'get_repo' do
+    action :create
+    endpoint new_resource.api_url
+    username new_resource.api_user
+    password new_resource.api_password
+
+    content <<EOS
+    import groovy.json.JsonSlurper
+    import groovy.json.JsonOutput
+
+    def params = new JsonSlurper().parseText(args)
```

Comment:
I don't think you need something else than the name, so you don't need the Slurper